### ISSUE TITLE
Add FlashAttention backward kernel

### ISSUE THREAD

**Issue by noahgift** [paiml/trueno#86]
## Summary

Add FlashAttention-style backward kernel for memory-efficient attention gradient computation.

## Background

Issue #85 implemented the core backward kernels needed for training. Standard attention backward can be composed from existing kernels (GEMM backward + Softmax backward), but FlashAttention backward is a memory optimization for long sequences.

## Specification

**FlashAttention Backward** avoids materializing the N×N attention matrix:
- Reuses LSE (log-sum-exp) values from forward pass
- Tiled processing to fit in shared memory
- Computes ∂L/∂Q, ∂L/∂K, ∂L/∂V in a single fused kernel

### Parameters
- \`q_ptr\`, \`k_ptr\`, \`v_ptr\`: Original inputs from forward
- \`o_ptr\`: Forward output
- \`lse_ptr\`: Log-sum-exp values from forward (for numerical stability)
- \`grad_o_ptr\`: Upstream gradient
- \`grad_q_ptr\`, \`grad_k_ptr\`, \`grad_v_ptr\`: Output gradients

### Mathematical Specification
```
Forward: O = softmax(QK^T / √d) @ V
Backward (fused, tiled):
  ∂L/∂V_j = Σ_i P_ij^T @ ∂L/∂O_i
  ∂L/∂Q_i = Σ_j (∂L/∂P_ij @ K_j) / √d
  ∂L/∂K_j = Σ_i (∂L/∂P_ij^T @ Q_i) / √d
where P = softmax(QK^T / √d) reconstructed from LSE
```

## Priority

Low - This is a memory optimization. Standard attention backward works for moderate sequence lengths using existing kernels.

## References
- FlashAttention paper: https://arxiv.org/abs/2205.14135
- FlashAttention-2: https://arxiv.org/abs/2307.08691
